src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssetEndpoints.cs

```diff
@@ -135,49 +136,56 @@ private List<StaticWebAssetEndpoint> CreateEndpoints(
                         Name = "Last-Modified",
                         Value = lastModified,
                     },
-                ];
+                    default
+                };
 
                 if (values.ContainsKey("fingerprint"))
                 {
                     // max-age=31536000 is one year in seconds. immutable means that the asset will never change.
                     // max-age is for browsers that do not support immutable.
-                    headers.Add(new() { Name = "Cache-Control", Value = "max-age=31536000, immutable" });
+                    headers[5] = new() { Name = "Cache-Control", Value = "max-age=31536000, immutable" };
                 }
                 else
                 {
                     // Force revalidation on non-fingerprinted assets. We can be more granular here and have rules based on the content type.
                     // These values can later be changed at runtime by modifying the endpoint. For example, it might be safer to cache images
                     // for a longer period of time than scripts or stylesheets.
-                    headers.Add(new() { Name = "Cache-Control", Value = !string.IsNullOrEmpty(cacheSetting) ? cacheSetting : "no-cache" });
+                    headers[5] = new() { Name = "Cache-Control", Value = !string.IsNullOrEmpty(cacheSetting) ? cacheSetting : "no-cache" };
+                }
+
+                var properties = new StaticWebAssetEndpointProperty[values.Count + (values.Count > 0 ? 2 : 1)];
```

Comment:
One alternative to preallocating the exact array size would be to always allocate the conservative amount. Then `StaticWebAssetEndpoint` uses `Memory<>` properties instead of arrays, and we provide the endpoint with slices that contain only the populated items.

This could potentially make the logic in this method less fragile. However, the code change is probably larger, and there might be other consequences of using `Memory<>` (for example, maybe it affects serialization), so I'm not sure it would be a net benefit. Just putting the idea out there.

Reply:
The other alternative that I have contemplated (but not implemented) is to bypass the array creation altogether and generate the serialized json string directly (that would definitely work for headers) and will work for properties for the few well-known cases that we know.

That avoids creating the array altogether and the json serialization later on